internal/provider/cluster_resource_test.go

```diff
@@ -1219,6 +1414,8 @@ func onDemandSingleRegionClusterWithLimitsStep(
 	planType client.PlanType,
 	requestUnitLimit int64,
 	storageMibLimit int64,
+	upgradeType *client.UpgradeTypeType,
```

Comment:
this pattern of adding new fields for each type that needs to be controlled on occasion doesn't scale well.  I'm open to suggestions here.   One thing I was considering is using a config object pattern similar to what is done for backup configuration: https://github.com/cockroachdb/terraform-provider-cockroach/blob/main/internal/provider/cluster_resource_test.go#L1912-L1917

Reply:
Sure, having a struct might be nice, where we can have default values that don't need to be set on every call.

Reply:
Okay, I've done some refactoring.  Let me know what you think.  I added it to a [later commit](https://github.com/cockroachdb/terraform-provider-cockroach/pull/252/commits/e95da011a8916523fa22648b85bdaa1bb75d6eee).

I could make further convenience functions to save additional space. I was thinking about something like:

```go
func provisionedClusterStep(
	clusterName string,
	config slsConfig,
) resource.TestStep {
    config.vcpus = ptr(6)
    return serverlessClusterStep(clusterName, client.PLANTYPE_STANDARD, config)
}
```

Reply:
The changes LGTM.